src/components/gridlines.ts

```diff
@@ -1,37 +1,119 @@
+      if (scale != null &&
+          ! QuantitativeScale.prototype.isPrototypeOf(scale) &&
+          ! Plottable.Scales.Category.prototype.isPrototypeOf(scale)) {
+        throw new Error(which + " needs to inherit from Scale.QuantitativeScale or Scales.Category.");
+      }
+    };
+
+    private _xScale: Scale<any, number>;
+    private _yScale: Scale<any, number>;
     private _xLinesContainer: d3.Selection<void>;
     private _yLinesContainer: d3.Selection<void>;
+    private _xTicks: () => ScaleDomainValues;
+    private _yTicks: () => ScaleDomainValues;
 
-    private _renderCallback: ScaleCallback<QuantitativeScale<any>>;
+    /**
+     */
+    private _xLinePosition: ILinePosition;
+    private _yLinePosition: ILinePosition;
 
+    private _renderCallback: ScaleCallback<Scale<any, number>>;
     /**
      * @constructor
-     * @param {QuantitativeScale} xScale The scale to base the x gridlines on. Pass null if no gridlines are desired.
-     * @param {QuantitativeScale} yScale The scale to base the y gridlines on. Pass null if no gridlines are desired.
+     *
+     * @param {Scale<any, number>} xScale The scale to base the x
+     * gridlines on. Can be a category or numeric scale. Pass null if
+     * no gridlines are desired.
+     *
+     * @param {Scale<any, number>} yScale The scale to base the y
+     * gridlines on. Can be a category or numeric scale. Pass null if
+     * no gridlines are desired.
      */
-    constructor(xScale: QuantitativeScale<any>, yScale: QuantitativeScale<any>) {
-      if (xScale != null && !(QuantitativeScale.prototype.isPrototypeOf(xScale))) {
-        throw new Error("xScale needs to inherit from Scale.QuantitativeScale");
-      }
-      if (yScale != null && !(QuantitativeScale.prototype.isPrototypeOf(yScale))) {
-        throw new Error("yScale needs to inherit from Scale.QuantitativeScale");
-      }
+    constructor(
+        xScale: QuantitativeScale<any> | Plottable.Scales.Category,
+        yScale: QuantitativeScale<any> | Plottable.Scales.Category) {
+
+      Gridlines.validateScale(xScale, "xScale");
+      Gridlines.validateScale(yScale, "yScale");
+
       super();
       this.addClass("gridlines");
       this._xScale = xScale;
       this._yScale = yScale;
       this._renderCallback = (scale) => this.render();
+
+      this._xLinePosition = "center";
+      this._yLinePosition = "center";
+
       if (this._xScale) {
+        this._xTicks = this._mkTicks(xScale);
         this._xScale.onUpdate(this._renderCallback);
       }
       if (this._yScale) {
+        this._yTicks = this._mkTicks(yScale);
         this._yScale.onUpdate(this._renderCallback);
       }
     }
 
+    /**
+     * Gets the xLinePosition value.
+     */
+    public xLinePosition(): ILinePosition;
+    /**
+     * Sets the xLinePosition value, which specifies whether lines are drawn in
+     * the center of category or after it.
+     *
+     * Valid options are "center" and "after". Note that this only makes sense
```

Comment:
Well 1) they don't have to and 2) they are different components that don't know about eachother. The other option would be to add the setting to the Scale, but that doesn't really make sense to add a setting that is only used in this context to the scale.